Infrastructure/Services/ParcelService.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Infrastructure.Interfaces;
+using Infrastructure.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.DataAccess
+{
+    public class ParcelService : IParcelService
+    {
+        private readonly DatabaseContext _dbContext;
+
+        public ParcelService(DatabaseContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<Parcel> GetByTrackingId(string trackingId)
+        {
+            try
+            {
+                var parcel = await _dbContext.Parcels.FirstOrDefaultAsync(x => x.TrackingNumber == trackingId);
+                if (parcel == null)
+                {
+                    return null;
+                }
+                return parcel;
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        public async Task<IList<Parcel>> GetAllByUserId(int id)
+        {
+            try
+            {
+                return await _dbContext.Parcels.Where(x => x.ShipperID == id).ToListAsync();
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        public async Task<IList<Parcel>> GetAll()
+        {
+            try
+            {
+                return await _dbContext.Parcels.ToListAsync();
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        public async Task Insert(Parcel newParcel)
+        {
+            try
+            {
+                await _dbContext.Parcels.AddAsync(newParcel);
+                await _dbContext.SaveChangesAsync();
+            }
+            catch
+            {
+                throw;
+            }
+        }
```

Comment:
I think this should atleast return Id/tracking number; tracking number would be best, as we will display it once the parcel is registered succesfullt